go/cmd/dolt/commands/push.go

```diff
@@ -264,18 +268,40 @@ func pushToRemoteBranch(ctx context.Context, srcRef, destRef, remoteRef ref.Dolt
 	return nil
 }
 
-func progFunc(progChan chan datas.PullProgress, stopChan chan struct{}) {
+func progFunc(progChan chan datas.PullProgress, stopChan chan struct{}, evt *events.Event) {
 	var latest datas.PullProgress
 	last := time.Now().UnixNano() - 1
 	lenPrinted := 0
 	done := false
+
+	timerMetricID := eventsapi.MetricID_DOWNLOAD_MS_ELAPSED
+	counterMetricID := eventsapi.MetricID_BYTES_DOWNLOADED
+
+	if evt.GetClientEventType() == eventsapi.ClientEventType_PUSH {
+		timerMetricID = eventsapi.MetricID_UPLOAD_MS_ELAPSED
+		counterMetricID = eventsapi.MetricID_BYTES_UPLOADED
+	}
+
+	counter := events.NewCounter(counterMetricID)
+	timer := events.NewTimer(timerMetricID)
+
+	var bytesWritten uint64
+
 	for !done {
 		select {
 		case progress, ok := <-progChan:
 			if !ok {
 				done = true
+
+				timer.Stop()
+				counter.Add(int32(bytesWritten))
+
+				evt.AddMetric(timer)
+				evt.AddMetric(counter)
 			}
+
 			latest = progress
+			bytesWritten = latest.ApproxWrittenBytes
 
```

Comment:
This is my main entry point for adding the `BYTES_UP/DOWNLOADED` metric and the `UP/DOWNLOAD_MS_ELAPSED` metric. As you can see I'm relying on `latest.ApproxWrittenBytes` to give me the bytes transferred, but need to find where I can get the exact number.